mockall/tests/automock_const_generic_struct_with_static_method.rs

```diff
@@ -0,0 +1,35 @@
+// vim: tw=80
+//! static non-generic methods of generic structs shouldn't require any special
+//! treatment when mocking.
+#![deny(warnings)]
+
+use mockall::*;
+use std::sync::{Mutex, PoisonError};
+
+static A_MTX: Mutex<()> = Mutex::new(());
+
+#[automock]
+trait A<const L: usize, const C: bool> {
+    fn bar() -> [u8; L];
+}
+
+#[test]
+fn generic_return() {
+    let _m = A_MTX.lock().unwrap_or_else(PoisonError::into_inner);
```

Comment:
Why the `PoisonError::into_inner`?  Also, this file mocks a `trait` not a `struct`, so the file name is wrong.